src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -129,9 +131,9 @@ private static boolean doSpin(final int numOfSpins) {
         boolean result = false;
         if (dev == null) {
             appendTrace("doSpin : Error, device is null");
-            return false;
+            result = false;
         }
-        if(isValidSpin(numOfSpins)) {
+        else if(isValidSpin(numOfSpins)) {
             for (int i = 0; i < numOfSpins; i++) {
```

Comment:
Here, instead of explicitly returning in this conditional branch, as klambo94 suggested, I have implemented a waterfall of if/else conditional branches and used the single return line.
I agree that this is easier to understand.